intercom-common/src/generators/cpp.rs

```diff
@@ -177,69 +184,103 @@ impl CppModel {
     ///
     /// - `path` - The path must point to a crate root containing Cargo.toml or
     ///            to the Cargo.toml itself.
-    pub fn from_path( path : &Path,) -> Result<CppModel, GeneratorError>
+    pub fn from_path(
+        path : &Path,
+        all_type_systems : bool,
+    ) -> Result<CppModel, GeneratorError>
     {
         let krate = model::ComCrate::parse_package( path )
                 .map_err( GeneratorError::CrateParseError )?;
-        CppModel::from_crate( &krate )
+        CppModel::from_crate( &krate, all_type_systems )
     }
 
 
     /// Converts the parse result into an header  that gets written to stdout.
     pub fn from_crate(
         c : &model::ComCrate,
+        all_type_systems : bool,
     ) -> Result<CppModel, GeneratorError> {
 
+        let itf_variant_filter : Box<Fn( &( &TypeSystem, &ComInterfaceVariant ) ) -> bool> =
+                match all_type_systems {
+                    true => Box::new( | _ | true ),
+                    false => Box::new( | ( ts, _ ) | match ts {
+                        TypeSystem::Raw => true,
+                        _ => false
+                    } ),
+                };
```

Comment:
"type_system_selector" instead of itf_variant_filter?

[VARIANT](https://docs.microsoft.com/en-us/windows/desktop/WinAuto/variant-structure) is a loaded term in COM world.